### PR TITLE
CONN-2959 Beta Config Based C2C

### PR DESCRIPTION
**Pull request by carlos-castillo-jask**
## PLEASE READ

Following a recent back-end update, all contributors with a local clone or fork of our repository are required to run `yarn install`. This does not apply to [direct page edits](https://help.sumologic.com/docs/contributing/edit-doc/#minor-edits).

- [x] Yes, I've run `yarn install`
- [ ] No, does not apply to me

## Purpose of this pull request

Provide a public beta doc link to internal and external users in order to retrieve feedback on a new general purpose C2C for collecting log data from simple APIs.

https://sumologic.atlassian.net/browse/CONN-2959

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions, .clabot
- [ ] Update Content - Revisions, updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

### CODE REVIEW

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
+---
+id: config-based-source
+title: Config Based Source
+sidebar_label: STIX/TAXII 2 Client
+tags:
+  - cloud-to-cloud
+  - config
+  - based
+description: Learn how to set up a Config Based C2C to collect data into the Sumo Logic environment.
+---
```
Comment (carlos-castillo-jask) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:10]: I've noticed the title does not appear and I don't know why.

image

**Review by kimsauce — APPROVED** [review]
good catches

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md (hunk 1 above)
Comment (kimsauce) [reply docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:10]: It looks like the author used one hash on Setup, which markdown reads as a title, and then itoverrides the actual title. So you just have to add a second hash and that'll fix it.

Screenshot 2024-03-07 at 1 43 09 PM

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+<p><a href="/docs/beta"><span className="beta">Beta</span></a></p>
+
+Sumo Logic provides this configuration based cloud source to collect log data from vendor APIs with a modular configuration. The goal of this source is for Sumo Logic to expand the configuration modules over time giving greater compatibility with vendor APIs, but to acknowledge complex APIs will still require a specific cloud source and not be compatible with this source.
+
+:::note
+This source is available in the [Fed deployment](/docs/api/getting-started#sumo-logic-endpoints-by-deployment-and-firewall-security).
+:::
+
+# Setup
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:27]:
```suggestion
## Setup
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+import CodeBlock from '@theme/CodeBlock';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+<head>
+  <meta name="robots" content="noindex" />
+</head>
+
+<p><a href="/docs/beta"><span className="beta">Beta</span></a></p>
+
+Sumo Logic provides this configuration based cloud source to collect log data from vendor APIs with a modular configuration. The goal of this source is for Sumo Logic to expand the configuration modules over time giving greater compatibility with vendor APIs, but to acknowledge complex APIs will still require a specific cloud source and not be compatible with this source.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:21]:
```suggestion
With our configuration-based (_config-based_) cloud source, you can collect log data from vendor APIs with a modular configuration. The goal of this source is for Sumo Logic to expand the configuration modules over time giving greater compatibility with vendor APIs, but to acknowledge complex APIs will still require a specific cloud source and not be compatible with this source.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+1. Locate the vendors API docs describing the API.
+1. Identify how the API implements authentication.
+1. Identify one API endpoint to collect logs from. You will need to create multiple Sumo sources if you wish to collect log data from more than one endpoint.
+   - What is the endpoint URL?
+   - Are there any required HTTP Headers?
+   - What URL parameters are available to use? Which ones track progression such as ingest timestamps?
+   - How is pagination implemented? 
+   - Where in the response body is the array of logs to ingest?
+   - Are there timestamps within the response array of logs indicating the individual log timestamp?
+1. Identify API rate limits
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:46]:
```suggestion
1. Identify API rate limits.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+   - Does the API specify a rate limit?
+   - Is bursting allowed?
+
+## Source Configuration
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:50]:
```suggestion
## Source configuration
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+   - ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema it is ignored, known as dropped.
+1. **Configuration Sections**. Expand each section to learn more about the options available for configuration. 
+<details>
+  <summary>Authentication</summary>
+  <div>
+    Choose the type of authentication the source will use based on the vendor API requirements and configure the details of that specific authentication type.
+
+#### Basic
+
+It is a default value. Select this authentication option if the vendor API requires basic HTTP authentication. The source will always add the `Authorization` HTTP request header with the base64 encoded `username:password` as the value.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:72]:
```suggestion
Basic is the default value. Select this authentication option if the vendor API requires basic HTTP authentication. The source will always add the `Authorization` HTTP request header with the base64 encoded `username:password` as the value.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+
+- **Basic HTTP Username**: The username required by the vendor API.
+- **Basic HTTP Password**: The password required by the vendor API.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:75]:
```suggestion
- **Basic HTTP Username**. The username required by the vendor API.
- **Basic HTTP Password**. The password required by the vendor API.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md (hunk 1 above)
Comment (kimsauce) [reply docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:10]: fixed

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md [hunk 8]
```diff
@@ -0,0 +1,468 @@
+---
+id: config-based-source
+title: Config Based Source
+sidebar_label: STIX/TAXII 2 Client
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md:4]:
```suggestion
sidebar_label: Config Based Source
```

**Review by kimsauce — APPROVED** [review]

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md (hunk 2 above)
Comment (kimsauce) [reply docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md]: fixed

**Pull request merged**